.github/workflows/verify_docker_image.yml

```diff
@@ -1,33 +1,114 @@
 name: Pull and Verify Docker Image
 
 on:
-  workflow_call:
   workflow_dispatch:
+    inputs:
+      amdgpu_target:
+        description: "AMDGPU target to verify"
+        required: true
+        default: "gfx942"
+        type: string
+      image_digest:
+        description: "Digest of image to verify"
+        required: true
+        type: string
+      runner_type:
+        description: "Runner type to use"
+        required: true
+        default: "linux-mi300-1gpu-ossci-rocm"
+        type: string
+      dispatch_id:
+        description: "Auto Generated Id for a dispatch"
+        type: string
+        required: true
+      version:
+        description: "Version Number as X.Y.ZrcYYYYMMDD format"
+        type: string
+        required: true
+  workflow_call:
+    inputs:
+      amdgpu_target:
+        type: string
+        required: true
+      image_digest:
+        type: string
+        required: true
+      runner_type:
+        type: string
+        required: true
+      dispatch_id:
+        type: string
+        required: true
+      version:
+        type: string
+        required: true
+
+permissions:
+  contents: read
+  packages: write
+
 jobs:
-  pull-and-build-image:
-    name: Verify Docker Image
-    runs-on: linux-mi300-1gpu-ossci-rocm
+  pull-and-verify-image:
+    name: Verify ${{ inputs.dispatch_id }} Docker Image
+    runs-on: ${{ inputs.runner_type }}
     steps:
+      - name: Identifier ${{ inputs.dispatch_id }}
+        run: |
+          echo "Dispatched for: ${{ inputs.dispatch_id }}"
+
       - name: Checkout repository
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Construct image reference
+        id: image_ref
+        env:
+          IMAGE_REF: "ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_${{ inputs.amdgpu_target }}@${{ inputs.image_digest }}"
+        run: |
+          echo "IMAGE_REF=$IMAGE_REF" >> "$GITHUB_ENV"
+          echo "Image to verify: $IMAGE_REF"
```

Comment:
If this is an environment variable the step is not required at all and you can use `${{ env.IMAGE_REF}}` instead.